### DISCUSSION TITLE
html5 starts with music muted

### DISCUSSION THREAD

**Discussion by yanargyr** [HaxeFlixel/flixel#2926] [Q&A] [answered]
When I export my game in html5, and when launching the game, there is no sound at all. If I hit the '+' button once, or the '0' button twice, the sounds are getting enabled. It looks like the html5 target always launches on mute, and the user needs to unmute manually.
It happens with any browser I've tried, and it only happens in html5 (was never a problem with cpp or mac)

I have even tried force unmutting on init like this :
```
 #if web
            FlxG.sound.muted = false;
            FlxG.sound.volume = 0.5;
        #end
```
But it still doesn't work

Is this normal behaviour, or something that you've seen before? Could it be a project setting I might have missed?
I am using HaxeFlixel 5.2.2

**Comment by Geokureli** [comment, marked as answer]
this is based on browser behavior and it's different for various browsers and also may depend on your specific browser's settings. Usually you need to click the game in order for sounds to be allowed to play, we don't really have control over this as it's a security feature

**Reply by yanargyr** [reply]
Yes, indeed I think the issue is related to focus.
I have further issues with this, as I noticed that when I lose focus (e.g. hit the address bar, which makes flx pause), and get back to the game, a new music playback is played, so if I unpause 10 times, I have 10 tracks playing over each other

I'm trying to work around this by using FLX_NO_FOCUS_LOST_SCREEN and ask the user to get the app foxus and hit the + before showing them the intro screen, but I was wondering if I missed something, or misconfigured etc, 

Thanks for your answer

**Reply by yanargyr** [reply]
OK, so I've worked around this, by doing this check in my Main.hx
```
#if web
    addChild(new FlxGame(gameWidth, gameHeight, Html5HackState, framerate, framerate, skipSplash, startFullscreen));
#else
    addChild(new FlxGame(gameWidth, gameHeight, MainMenuState, framerate, framerate, skipSplash, startFullscreen));
#end
```

And my Html5HackState just wait for a player interaction, make sure I have focus, and plays a sound that unmutes the sounds and moves on the the normal app

```
package states;
import flixel.FlxG;
import flixel.text.FlxText;
import flixel.FlxState;

class Html5HackState extends FlxState {
    public function new() {
        super();
    }

    override public function create():Void {
        super.create();

        var text: FlxText = new FlxText(40,160, 200, "Press A to start the game");
        this.add(text);
    }

    override public function update(elapsed:Float):Void {
        super.update(elapsed);
        if (
            FlxG.keys.anyJustPressed(["A"])
        ){
            FlxG.sound.play("assets/sounds/jump.ogg", 0);
            FlxG.switchState(new MainMenuState());
        }
    }
}
```

**Reply by Geokureli** [reply]
> I have further issues with this, as I noticed that when I lose focus (e.g. hit the address bar, which makes flx pause), and get back to the game, a new music playback is played, so if I unpause 10 times, I have 10 tracks playing over each other

I've never seen this behavior, what browser are you using?


Also:
```hx
FlxG.keys.anyJustPressed(["A"])
```
weird pet-peeve, why do it this way when you can do it without quotes? `FlxG.keys.anyJustPressed([A])` A without quotes shows up in type hinting and it throws compile errors if you put a typo, or a key that isn't valid. also there's:
```hx
FlxG.keys.justPressed.A
```

**Reply by yanargyr** [reply]
I reproduced the behavior with firefox and chrome
Thanks for the suggestions

**Reply by Geokureli** [reply]
you reproduced the stacked sound issue on those? interesting, can you share any relevant code? specifically interested in how you play the sound, when you play it, whether you're using `FLX_NO_FOCUS_LOST_SCREEN` or [FlxG.autoPause](https://api.haxeflixel.com/flixel/FlxG.html#autoPause) or if you're listening for focus events anywhere

**Reply by yanargyr** [reply]
@Geokureli  
I've created a simplified version for you here : https://github.com/yanargyr/FlixelWebSoundBugDemo 

I think the file size might be a factor, as I was using a 15 MB file in my case

If you hit the +/- buttons, the music for some reason starts properly and will not stack. 
But if you refresh, and instead of playing with the volume keys, you pause/unpause, the music will start playing. If you pause/resume again, you will get a second music channel added. Not quite sure it is actually adding a new music object each time, maybe it's just two.

If you want to repro with my actual game, you can also try here: https://yanargyr.github.io/ (that's before my workaround)

Hope that helps, and thanks again

**Reply by Geokureli** [reply]
I'll look into this today, thanks

**Reply by yanargyr** [reply]
Just in case that helps, I suspect that the music is initialised correctly if a sound plays first (SoundFrontEnd#play before SoundFrontEnd#playMusic). Hitting the  +/- keys does make a sound effect
